supplierProduct/tests.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+from supplierProduct.models import supplierProduct, Product, Supplier
+
+supID0 = 123
+supID1 = 222
+supID2 = 127
+
+
+class TestSupProduct:
+
+    @pytest.fixture
+    def creat_product(self):
+        prod0 = Product(qr_code="AASSD22", product_name="APPLE", description="sweet!!")
+        prod1 = Product(qr_code="QS#$HHS", product_name="CARROT", description="yummy!!")
+        prod2 = Product(qr_code="AAsdgv66", product_name="BANANA", description="great!!")
+        prod0.save()
+        prod1.save()
+        prod2.save()
```

Comment:
product has a function called saved product,You here not using it and I can see that qr code is wrong and not contains 32 chars

Reply:
but shouldn't the qr code be max 32 chars ?

Reply:
Nope,it is fixed look at the validators